lib/wave_gen/wave_gen.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdlib.h>
+
+#include <math.h>
+#ifndef M_PI
+  #define M_PI 3.14159265358979323846
+#endif
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wave_gen, CONFIG_WAVE_GEN_LIB_LOG_LEVEL);
+
+#include <wave_gen.h>
+
+/**
+ * @brief Generates a pseudo-random number between -1 and 1.
+ */
+static double generate_pseudo_random(void)
+{
+	return (double)rand() / ((double)RAND_MAX / 2.0) - 1.0;
```

Comment:
2.0 is double so compiler will handle casting of RAND_MAX, then it will expand rand() result if needed

Reply:
Copy-paste from sensor-sim driver. Fixed